Backend/endpoints/filesystem_endpoints.go

```diff
@@ -203,3 +204,67 @@ func mapOver[T any, V any](input []T, mapFunction func(T) V) []V {
 	}
 	return output
 }
+
+type ValidImageUploadRequest struct {
+	Parent      int
+	LogicalName string `schema:"LogicalName,required"`
+	OwnerGroup  int    `schema:"OwnerGroup,required"`
+}
+
+func UploadImage(w http.ResponseWriter, r *http.Request, df DependencyFactory, log *logger.Log) (int, interface{}, error) {
+	var input ValidImageUploadRequest
+	if status := ParseMultiPartFormToSchema(r, "POST", &input); status != http.StatusOK {
+		return status, nil, nil
+	}
+
+	// Parse multipart file, with max size of 10 MB
+	// Extract image and check for error
+	uploadedFile, _, err := r.FormFile("Image")
+	if err != nil {
+		log.Write([]byte("Error retrieving the file"))
+		return http.StatusBadRequest, nil, err
+	}
+	defer uploadedFile.Close()
+
+	// Create entity in repository
+	fs := reflect.TypeOf((*repositories.IFilesystemRepository)(nil))
+	dfs := reflect.TypeOf((*repositories.IDockerUnpublishedFilesystemRepository)(nil))
+
+	repository := df.GetDependency(fs).(repositories.IFilesystemRepository)
+	dockerRepository := df.GetDependency(dfs).(repositories.IDockerPublishedFilesystemRepository)
+	log.Write([]byte("Acquired repository."))
+
+	entityToCreate := repositories.FilesystemEntry{
+		LogicalName:  input.LogicalName,
+		ParentFileID: input.Parent,
+		IsDocument:   false,
+		OwnerUserId:  input.OwnerGroup,
+	}
+
+	e, err := repository.CreateEntry(entityToCreate)
+	if err != nil {
+		log.Write([]byte("Failed request!"))
+		return http.StatusNotAcceptable, nil, err
+	}
+
+	// Create and get a new entry in docker file system
+	dockerRepository.AddToVolume(strconv.Itoa(e.EntityID))
+	log.Write([]byte(fmt.Sprintf("Created new entity %v.", entityToCreate)))
+	dockerFile, err := dockerRepository.GetFromVolume(strconv.Itoa(e.EntityID))
+	if err != nil {
+		log.Write([]byte("Unable to get docker file"))
```

Comment:
Nitpick: You dont have a fullstop here but you have one everywhere else :P

Reply:
Guess I'll die